starlette/datastructures.py

```diff
@@ -604,6 +604,21 @@ def __delitem__(self, key: str) -> None:
         for idx in reversed(pop_indexes):
             del self._list[idx]
 
+    def __ior__(self, other: typing.Union[dict, "MutableHeaders"]) -> "MutableHeaders":
+        if isinstance(other, MutableHeaders):
+            other = dict(other)
+        for key, val in other.items():
+            self[key] = val
+        return self
+
+    def __or__(self, other: typing.Union[dict, "MutableHeaders"]) -> "MutableHeaders":
+        new = self.mutablecopy()
+        if isinstance(other, MutableHeaders):
+            new.update(dict(other))
+        else:
+            new.update(other)
+        return new
```

Comment:
why not just implement them in terms of `update()` and forego the typechecking?

```
    def __ior__(self, other: typing.Union[dict, "MutableHeaders"]) -> "MutableHeaders":
        self.update(other)
        return self

    def __or__(self, other: typing.Union[dict, "MutableHeaders"]) -> "MutableHeaders":
        new = self.mutablecopy()
        new.update(other)
        return new
```

Reply:
@jcwilson Absolutely right. Changes have been applied. Thank you.